### ISSUE TITLE
inference error  using onnxruntime

### ISSUE THREAD

**Issue by oym050922021** [open-mmlab/mmdeploy#2127]
### Checklist

- [X] I have searched related issues but cannot get the expected help.
- [X] 2. I have read the [FAQ documentation](https://github.com/open-mmlab/mmdeploy/tree/main/docs/en/faq.md) but cannot get the expected help.
- [X] 3. The bug has not been fixed in the latest version.

### Describe the bug

I used onnxruntime to infer the yolox model and got incorrect results.  when I used the mmdeploy API named mmdeploy.apis.inference_model to infer the model, I can get the correct results. What's the reason , please?
If i don't want to use the mmdeploy API, is there any inference example with onnxruntime?

### Reproduction

####基于onnxruntime做推理
import cv2
import numpy as np
import torch
from PIL import Image
import torchvision.transforms as transforms
import onnxruntime

def preprocess(image_path):

    transform = transforms.Compose([transforms.ToTensor(),
                                    transforms.Resize((640, 640))
                                    ])
    image = Image.open(image_path)
    image = transform(image)
    image = image.numpy().astype(np.float32)
    image = np.expand_dims(image, axis = 0)
    return image

if __name__ == "__main__":
    device = torch.device('cpu')
    image_path = "/root/Ouyangmei/openmmlab/mmdetection/datasets/teeth/images/test/0128.png"
    model_path = "/root/Ouyangmei/openmmlab/mmdeploy/mmdeploy_models/mmdeploy/yolox/end2end.onnx"
    img = preprocess(image_path)
    ort_session = onnxruntime.InferenceSession(model_path, providers=['CPUExecutionProvider'])
    ort_inputs = {ort_session.get_inputs()[0].name: img}
    ort_outputs = ort_session.run(None, ort_inputs)
    labels_numpy = np.squeeze(np.array(ort_outputs[1]),axis=0)
    bboxes_score_numpy = np.squeeze(np.array(ort_outputs[0]), axis =0)

    for box_ind in range(len(bboxes_score_numpy)):
        bbox = bboxes_score_numpy[box_ind,:-1]
        scores = bboxes_score_numpy[box_ind,-1]
        label = labels_numpy[box_ind]
        x1,y1,x2,y2 = bbox

### Environment

```Shell
the prediction results are all zeros.
```

**Comment by irexyc** [comment]
You should normalize the image.

Please take a look at :
https://github.com/open-mmlab/mmdetection/blob/main/configs/yolox/yolox_s_8xb8-300e_coco.py#L11-L20
https://github.com/open-mmlab/mmdetection/blob/main/mmdet/models/data_preprocessors/data_preprocessor.py#L31

**Comment by oym050922021** [comment]
Hi, @irexyc 
 When I normalize the image, do I need to normalize according to the test-pipeline and  data_preprocessor in the configs? Thanks.

**Comment by irexyc** [comment]
@oym050922021 
Yes, to make your results same with pytorch. You should follow the preprocess of you pytorch model.

MMDetection 3.x moves some transforms like normalization to data_preprocess of model.

**Comment by oym050922021** [comment]
@irexyc ok, thank you. I have solved the problem.
